spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java

```diff
@@ -0,0 +1,233 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.dataflow.core.ModuleDeploymentId;
+import org.springframework.cloud.dataflow.core.ModuleDeploymentRequest;
+import org.springframework.cloud.dataflow.module.ModuleInstanceStatus;
+import org.springframework.cloud.dataflow.module.ModuleStatus;
+import org.springframework.cloud.dataflow.module.deployer.ModuleArgumentQualifier;
+import org.springframework.cloud.dataflow.module.deployer.ModuleDeployer;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.util.FileCopyUtils;
+import org.springframework.util.SocketUtils;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * A ModuleDeployer implementation that spins off a new JVM process per
+ * module instance.
+ *
+ * @author Eric Bottard
+ */
+public class OutOfProcessModuleDeployer implements ModuleDeployer {
+
+	private String moduleLauncherPath;
+
+	private static final Logger logger = LoggerFactory.getLogger(OutOfProcessModuleDeployer.class);
+
+	private Map<ModuleDeploymentId, List<Instance>> running = new HashMap<>();
+
+	@Autowired
+	private OutOfProcessModuleDeployerProperties properties;
+
+	@Override
+	public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
+
+		String module = request.getCoordinates().toString();
+
+		ModuleDeploymentId moduleDeploymentId = ModuleDeploymentId.fromModuleDefinition(request.getDefinition());
+		List<Instance> processes = new ArrayList<>();
+		running.put(moduleDeploymentId, processes);
+
+
+		HashMap<String, String> args = new HashMap<>();
+		args.put("modules", request.getCoordinates().toString());
+		args.putAll(ModuleArgumentQualifier.qualifyArgs(0, request.getDefinition().getParameters()));
+		args.putAll(ModuleArgumentQualifier.qualifyArgs(0, request.getDeploymentProperties()));
+		String jmxDomainName = String.format("%s.%s", request.getDefinition().getGroup(), request.getDefinition().getLabel());
+		args.putAll(ModuleArgumentQualifier.qualifyArgs(0, Collections.singletonMap(JMX_DEFAULT_DOMAIN_KEY, jmxDomainName)));
+
+		args.put("endpoints.shutdown.enabled", "true");
+		args.put("endpoints.jmx.unique-names", "true");
+
+
+		for (int i = 0; i < request.getCount(); i++) {
+			int port = SocketUtils.findAvailableTcpPort(DEFAULT_SERVER_PORT);
+			args.put(SERVER_PORT_KEY, String.valueOf(port));
+
+			ProcessBuilder builder = new ProcessBuilder(properties.getJavaCmd(), "-jar", moduleLauncherPath);
+			builder.environment().clear();
+			builder.environment().putAll(args);
+
+			try {
+				Path workDir = Files.createTempDirectory(properties.getWorkingDirectoriesRoot(), moduleDeploymentId.toString() + "-" + i + "-");
+				if (properties.isDeleteFilesOnExit()) {
+					workDir.toFile().deleteOnExit();

```

Comment:
This requires registering files for deletion too: see below.

As an improvement - I think that we can create a single directory per module deployment (rather than per instance) and create separate files per instance id inside that directory - thus being able to do something like `tail -f stdout_*` to aggregate the output of multiple modules.